### PR TITLE
fix: disable country chip when field is disabled

### PR DESCRIPTION
**Pull request by sidlatau**
Before change:
image

After change:
image

### CODE REVIEW

**Comment by cedvdb** [comment]
Could you add a small test to ensure this does not break in the future ?

**Comment by sidlatau** [comment]
> Could you add a small test to ensure this does not break in the future ?

@cedvdb I added a couple of tests.

File: test/phone_form_field_test.dart [hunk 1]
```diff
@@ -62,6 +64,20 @@ void main() {
         await tester.pumpWidget(getWidget());
         expect(find.byType(CircleFlag), findsWidgets);
       });
+
+      testWidgets('enabled, country chip should be enabled', (tester) async {
+        await tester.pumpWidget(getWidget(enabled: true));
+        final countryChip =
+            tester.widget<CountryCodeChip>(find.byType(CountryCodeChip));
+        expect(countryChip.enabled, true);
```
Comment (cedvdb) [inline test/phone_form_field_test.dart:72]: We probably want to test that when disabled, tapping on the country code does not do anything. 
 
Currently those two test do not really cover much, so are, imo, unnecessary.

**Comment by cedvdb** [comment]
Thanks a lot

**Pull request merged**